### ISSUE TITLE
Cassandra host could not be resolved

### ISSUE THREAD

**Issue by montanaflynn** [Mashape/kong#195]
Someone in gitter reported that Kong could not connect to a kongdb.org Cassandra node, to make sure it wasn't kongdb we set them up with a new Cassandra node on AWS and they got the same error. 

They were running Ubuntu 15.04 with Kong 0.2.0-2  but I [verified it works](https://gist.github.com/montanaflynn/42a32c1d7844e8038fd6) on a brand new AWS machine so there's something else at play here, possibly related to #127 .

The initial migrations worked and connection with the host was verified with:

```
telnet kongdb.org 48134
Trying 52.5.149.55...
Connected to kongdb.org.
```

Starting up Kong:
```
kong start
[INFO] Using configuration: /etc/kong/kong.yml
[INFO] Proxy port.........8000
Admin API port.....8001
Database...........cassandra keepalive=60000 port=48134 timeout=1000 hosts=kongdb.org keyspace=kong
[WARN] ulimit is currently set to "1024". For better performance set it to at least "4096" using "ulimit -n"
[OK] Started
```

Trying to use Kong's admin API:
```
curl 127.0.0.1:8001/
{"message":"Cassandra error: kongdb.org could not be resolved (110: Operation timed out)"}
```

**Comment by pgeseraphin** [comment]
In kong.yml replace "kongdb.org" by "52.5.149.55" in hosts section of Cassandra properties. It should work.

**Comment by montanaflynn** [comment]
Thanks @pgeseraphin, I can't believe we didn't think of that! I'll send the person who initially reported the issue the solution and follow up with an update here if it works.

**Comment by williamn** [comment]
@montanaflynn @pgeseraphin I just try to us IP instead of hostname. It didn't work
```
$ sudo kong start            
[INFO] Using configuration: /etc/kong/kong.yml
[INFO] Proxy port.........8000
       Admin API port.....8001
       Database...........cassandra keepalive=60000 port=34474 timeout=1000 hosts=52.5.149.55 keyspace=kong
[ERR] Cassandra error: connection refused
```

**Comment by pgeseraphin** [comment]
Don't forget to change Cassandra port in 'kong.yml'. You'll get it with: 
curl -X POST http://kongdb.org/provision

**Comment by williamn** [comment]
@pgeseraphin ah you right. I forgott that.

**Comment by williamn** [comment]
@montanaflynn @pgeseraphin it works when I change to IP instead of host name. Thanks.

**Comment by pgeseraphin** [comment]
You're welcome.

**Comment by williamn** [comment]
I still got error when I forward my request through Kong

```
$ curl -i -X GET \
 --url http://localhost:8000/ \
 --header 'Host: mockbin.com'
HTTP/1.1 502 Bad Gateway
Server: openresty/1.7.10.1
Date: Mon, 04 May 2015 08:58:34 GMT
Content-Type: text/html; charset=UTF-8
Content-Length: 179
Connection: keep-alive
Via: kong/0.2.0-2

<html>
<head><title>502 Bad Gateway</title></head>
<body bgcolor="white">
<center><h1>502 Bad Gateway</h1></center>
<hr><center>openresty/1.7.10.1</center>
</body>
</html>
```

**Comment by pgeseraphin** [comment]
It's another issue (see #182). To solve this, when adding an API, you have to use '127.0.0.1' instead of 'localhost':

Example:

$ curl -i -X POST \
 --url http://127.0.0.1:8001/apis/ \
 --data 'name=mockbin' \
 --data 'target_url=http://mockbin.com/' \
 --data 'public_dns=mockbin.com'

**Comment by williamn** [comment]
@pgeseraphin still not working

```
curl -i -X POST \
>  --url http://127.0.0.1:8001/apis/ \
>  --data 'name=mockbin' \
>  --data 'target_url=http://mockbin.com/' \
>  --data 'public_dns=mockbin.com'
HTTP/1.1 201 Created
Date: Mon, 04 May 2015 09:21:56 GMT
Content-Type: application/json
Transfer-Encoding: chunked
Connection: keep-alive
Server: kong/0.2.0-2

{"public_dns":"mockbin.com","target_url":"http:\/\/mockbin.com\/","id":"0b395813-d65b-4cbf-cca5-4a711202ff08","created_at":1430731313000,"name":"mockbin"}
```

When I request

```
curl -i -X GET \                                                                       
 --url http://localhost:8000/ \
 --header 'Host: mockbin.com'
HTTP/1.1 502 Bad Gateway
Server: openresty/1.7.10.1
Date: Mon, 04 May 2015 09:22:42 GMT
Content-Type: text/html; charset=UTF-8
Content-Length: 179
Connection: keep-alive
Via: kong/0.2.0-2

<html>
<head><title>502 Bad Gateway</title></head>
<body bgcolor="white">
<center><h1>502 Bad Gateway</h1></center>
<hr><center>openresty/1.7.10.1</center>
</body>
</html>
```

**Comment by pgeseraphin** [comment]
Try to replace http://mockbin.com/ by its IP: http://74.208.32.158/ or use your own API.

$ curl -i -X POST \
--url http://127.0.0.1:8001/apis/ \
--data 'name=mockbin' \
--data 'target_url=http://74.208.32.158/' \
--data 'public_dns=mockbin.com'

**Comment by thefosk** [comment]
This issue might should been fixed with #194 - in order to work it needs `dnsmasq

**Comment by thefosk** [comment]
@williamn were you the initial user who was affected by this bug?

**Comment by williamn** [comment]
@thefosk yes

**Comment by williamn** [comment]
It works after I install `dnsmasq`

**Comment by thefosk** [comment]
@williamn that's great. We [just released v0.2.1](https://github.com/Mashape/kong/releases/tag/0.2.1), which includes `dnsmasq` by default.

I will close this for now, please feel free to re-open it if you experience again the same problem.

**Comment by williamn** [comment]
:beers:
